src/metabase/driver/sql_jdbc/sync/interface.clj

```diff
@@ -27,6 +27,13 @@
   driver/dispatch-on-initialized-driver
   :hierarchy #'driver/hierarchy)
 
+(defmulti have-any-schema-privileges?
+  "Check if we have any sort of privileges for the given `schema`. If this method is defined for some driver, and it
+  returns false for the given schema, then we won't attempt to check for any tables within."
+  {:arglists '([driver ^java.sql.Connection connection ^String table-schema])}
+  driver/dispatch-on-initialized-driver
+  :hierarchy #'driver/hierarchy)
```

Comment:
I think you should change this to something like `remove-schemas-without-privileges` so drivers that can determine them all at once can do so... checking 1000 schemas one-at-a-time is potentially a little slow, so it would be nice to give drivers the chance to optimize it a bit.

Better yet... why not just make `syncable-schemas` the new multimethod and let drivers filter out things however they please? Maybe they're filtering out ones based on some predefined blacklist, or maybe they're doing perms checking like Redshift does. The default impl can still use `excluded-schemas`

Reply:
OK, given everything outlined above, I think it makes the most sense to just have `syncable-schemas` as the multimethod, and duplicate a tiny bit in building up the `eduction` for Redshift.